### ISSUE TITLE
Build failes because rocrand is missing, but it is installed

### ISSUE THREAD

**Issue by jaschaknack** [rocm-arch/python-pytorch-rocm#38]
Trying to install python-pytorch-rocm, rocrand is installed but not found.

```
***** Library versions from cmake find_package *****

-- hip::amdhip64 is SHARED_LIBRARY
hip VERSION: 5.1.22140
hsa-runtime64 VERSION: 1.5.0
amd_comgr VERSION: 2.4.0
CMake Error at cmake/public/LoadHIP.cmake:147 (find_package):
  By not providing "Findrocrand.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "rocrand", but
  CMake did not find one.

  Could not find a package configuration file provided by "rocrand" with any
  of the following names:

    rocrandConfig.cmake
    rocrand-config.cmake

  Add the installation prefix of "rocrand" to CMAKE_PREFIX_PATH or set
  "rocrand_DIR" to a directory containing one of the above files.  If
  "rocrand" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  cmake/public/LoadHIP.cmake:252 (find_package_and_print_version)
  cmake/Dependencies.cmake:1238 (include)
  CMakeLists.txt:669 (include)


-- Configuring incomplete, errors occurred!
See also "/tmp/makepkg/python-pytorch-rocm/src/pytorch-1.11.0-rocm/build/CMakeFiles/CMakeOutput.log".
See also "/tmp/makepkg/python-pytorch-rocm/src/pytorch-1.11.0-rocm/build/CMakeFiles/CMakeError.log".
==> ERROR: A failure occurred in build().
    Aborting...
 -> error making: python-pytorch-rocm
```

**Comment by navti** [comment]
Those config cmake files are present in rocrand installation folder. For e.g., my rocrand installation folder is at 
`/opt/rocm/rocrand` 
and the config cmake file is at 
`/opt/rocm/rocrand/lib/cmake/rocrand/rocrand-config.cmake`
Check if your rocrand installation has that config file.

**Comment by mikesbytes** [comment]
Having the same issue, there is no `lib` folder in `/opt/rocm/rocrand`

**Comment by navti** [comment]
> Having the same issue, there is no `lib` folder in `/opt/rocm/rocrand`

May I know how did you install rocrand? Did you do it from AUR or installed pre built binaries from a third party repo?

**Comment by JP-Ellis** [comment]
I think this might be related to deprecation of
```
#include <rocrand.h>
```
in favour of
```
#include <rocrand/rocrand.h>
```
Up until the last release (<5.1.1), the backwards compatibility was kept though in 5.1.1 they removed that.  It looks like it'll be reinstated in 5.2.0 though.

I have found that adding the following at the end of the `prepare` section fixes it (though I am still waiting on the build to finish).
```
export ROCRAND_PATH=/opt/rocm
export HIPRAND_PATH=/opt/rocm
```

**Comment by navti** [comment]
> I think this might be related to deprecation of
> 
> ```
> #include <rocrand.h>
> ```
> 
> in favour of
> 
> ```
> #include <rocrand/rocrand.h>
> ```
> 
> Up until the last release (<5.1.1), the backwards compatibility was kept though in 5.1.1 they removed that. It looks like it'll be reinstated in 5.2.0 though.
> 
> I have found that adding the following at the end of the `prepare` section fixes it (though I am still waiting on the build to finish).
> 
> ```
> export ROCRAND_PATH=/opt/rocm
> export HIPRAND_PATH=/opt/rocm
> ```

Yes, quite possible. I recently updated all my rocm packages to 5.1.1. But I installed pre built binaries from a third party repo so didn't have to build anything.

**Comment by mikesbytes** [comment]
Managed to build with the lines suggested by @JP-Ellis. Seems to have built successfully, but trying to use a a `cuda` device errors out with `RuntimeError: HIP error: hipErrorLaunchFailure`. Most likely a different issue though.

**Comment by navti** [comment]
> Managed to build with the lines suggested by @JP-Ellis. Seems to have built successfully, but trying to use a a `cuda` device errors out with `RuntimeError: HIP error: hipErrorLaunchFailure`. Most likely a different issue though.

Okay. That seems to be a different problem. Please close this issue and open a new one for this problem.

**Comment by zesterer** [comment]
Hi, I have the same error as the OP.

I've checked that I have the lines suggested by @JP-Ellis added to my `PKGBUILD`'s `prepare` section. I also have rocm installed (as far as I'm aware).

However, I do have the suggested `.cmake` files in `/opt/rocm/lib/cmake/rocrand`.

**Comment by zesterer** [comment]
> @zesterer I tried to add these 3 lines to the `PKGBUILD` and it fails for me the very same way. I can not make the symlink though, because both directories are not empty on the start.

Do you have ROCM properly installed, with the appropriate files `.cmake` in the locations I list in `CMAKE_PREFIX_PATH`?

**Comment by zaskokus** [comment]
@zesterer I've only followed the https://github.com/rocm-arch/rocm-arch instructions. Should I rebuild the whole codebase?

**Comment by zesterer** [comment]
I'm not sure, sorry. This is beyond me.

**Comment by acxz** [comment]
> It looks like it'll be reinstated in 5.2.0 though.

@JP-Ellis can you point to something upstream that clarifies this?

> export CMAKE_PREFIX_PATH="/opt/rocm/lib/cmake/rocrand;/opt/rocm/lib/cmake/hiprand;/opt/rocm/lib/cmake/miopengemm;${CMAKE_PREFIX_PATH}"

@zesterer hopefully this shouldn't be need with the latest python-pytorch-rocm-1.12.0 version. Can you try to rebuild the package and check if you are still running into this issue?

> (the latter two lines are because I have GCC 12 installed, but ROCm seems to be compatible with only GCC 11)

@zesterer this has been fixed in the latest PKGBUILD, python-pytorch-rocm now uses gcc-11

> my GPU is too old and the build fails with a message about a missing LLVM intrinsic

@zesterer please post another issue about this

**Comment by acxz** [comment]
I don't seem to run into this issue anymore. If anyone else can chime in that'd be great so that we can close this ticket.

**Comment by JP-Ellis** [comment]
> > It looks like it'll be reinstated in 5.2.0 though.
> 
> @JP-Ellis can you point to something upstream that clarifies this?

Sure, it's actually in the [`rocRAND` changelog file](https://github.com/ROCmSoftwarePlatform/rocRAND/blob/develop/CHANGELOG.md).

**Comment by acxz** [comment]
Sweet, closing this issue with the upgrade since arch's rocrand is at 5.2.0 now.
